chain/vm/runtime.go

```diff
@@ -222,6 +225,23 @@ func (rt *Runtime) GetActorCodeCID(addr address.Address) (ret cid.Cid, ok bool)
 		panic(aerrors.Fatalf("failed to get actor: %s", err))
 	}
 
+	// required for genesis/testing
+	if nv := rt.NetworkVersion(); nv >= network.Version16 {
+		name, av, ok := actors.GetActorMetaByCode(act.Code)
+
+		if ok {
+			// lies, lies, lies
+			builder := cid.V1Builder{Codec: cid.Raw, MhType: mh.IDENTITY}
+			synthetic := fmt.Sprintf("fil/%d/%s", av, name)
+			syntheticCid, err := builder.Sum([]byte(synthetic))
+			if err != nil {
+				panic(aerrors.Fatalf("failed to generate synthetic CID: %s", err))
+			}
+
+			return syntheticCid, true
+		}
+	}
```

Comment:
Really I think we should be trying to get rid of the `id(fil/8/y)` code cids from the codebase, otherwise we'll have to translate them back and forth everywhere, which will be really painful, especially with more network upgrades

Reply:
yes, I totally agree, but it's not as simple -- we'll have to do some surgery in specs-actors.

Reply:
Won't we remove most Go specs-actors logic, and mostly just keep the data-accessing bits? That should remove almost all dependencies on old code CIDs. (I'm assuming that builtin-actors don't use them - didn't look at them much yet)

Reply:
if we remove all specs-actors logic we will be fine; really, we should become fvm only, except for state.